### ISSUE TITLE
Altering instance defaults after creation seem to affect other instances as well

### ISSUE THREAD

**Issue by hontas** [axios/axios#1387]
#### Summary

Changing instance after creation like it says in the README alter other instances behaviour as well.

```js
const api = axios.create({ baseURL });
const authApi = axios.create({ baseURL });
authApi.defaults.headers.common['Authorization'] = `Bearer: ${token}`;
```

Doing a request with `api` now get the header of authApi causing that request to be authorized when it shouldn't be.

The following way works as expected:
```js
const api = axios.create({ baseURL });
const authApi = axios.create({ baseURL, headers: { Authorization: `Bearer: ${token}` } });
```

- axios version: v0.18.0
- Environment: node v8.9.4

**Comment by heisian** [comment]
Problem is the `utils.merge` function passes by reference on the nested `headers` object. Trying to get #1391 pushed through.

**Comment by emilyemorehouse** [comment]
Closing, as this issue was resolved in #1395. 🙌
